test/base_system_registration_dialog_test.rb

```diff
@@ -90,6 +90,23 @@
         end
       end
 
+      context "when user enters an invalid regcode" do
+        # include CRLF characters which are not allowed
+        let(:reg_code) { "\nmy-reg-code\r" }
+        it "displays error popup and does not register the system" do
+          expect(Yast::UI).to receive(:QueryWidget).with(:reg_code, :Value)
+            .and_return(reg_code)
+
+          expect(Yast::UI).to receive(:UserInput).and_return(:next, :abort)
+          expect(Registration::UI::AbortConfirmation).to receive(:run).and_return(true)
+
+          expect(Yast::Report).to receive(:Error).with(/Invalid registration code/)
```

Comment:
This (and the next line) is what you really want to test. So it is fine to use `expect` here.